astropy/uncertainty/core.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+from astropy.utils.misc import isiterable
+
+__all__ = ["Variable"]
+
+# Derivatives of ufuncs relative to their input(s).
+UFUNC_DERIVATIVES = {
+    np.add: (lambda x, y: 1.,
+             lambda x, y: 1.),
+    np.subtract: (lambda x, y: 1.,
+                  lambda x, y: -1.),
+    np.multiply: (lambda x, y: y,
+                  lambda x, y: x),
+    np.true_divide: (lambda x, y: 1/y,
+                     lambda x, y: -x/y**2),
+    np.arccos: lambda x: -1/np.sqrt(1-x**2),
+    np.arccosh: lambda x: 1/np.sqrt(x**2-1),
+    np.arcsin: lambda x: 1/np.sqrt(1-x**2),
+    np.arcsinh: lambda x: 1/np.sqrt(1+x**2),
+    np.arctan: lambda x: 1/(1+x**2),
+    np.arctan2: (lambda y, x: x/(x**2+y**2),
+                 lambda y, x: -y/(x**2+y**2)),
+    np.arctanh: lambda x: 1/(1-x**2),
+    np.copysign: (lambda x, y: np.where(x>=0, np.copysign(1, y),
+                                        -np.copysign(1, y)),
+                  lambda x, y: 0),
+    np.cos: lambda x: -np.sin(x),
+    np.cosh: np.sinh,
+    np.degrees: lambda x: np.degrees(1.),
+    np.exp: np.exp,
+    np.expm1: np.exp,
+    np.fabs: lambda x: np.where(x>=0., 1., -1.),
+    np.hypot: (lambda x, y: x/np.hypot(x, y),
+               lambda x, y: y/np.hypot(x, y)),
+    np.log: lambda x: 1/x,
+    np.log10: lambda x: 1/x/np.log(10),
+    np.log1p: lambda x: 1/(1+x),
+    np.power: (lambda x, y: x**(y-1) * y,   # explicitly zero y==0?
+               lambda x, y: np.log(x)*x**y),
+    # above: explicityly zero np.logical_and(x==0, y>0)?
+    np.radians: lambda x: np.radians(1),
+    np.sin: np.cos,
+    np.sinh: np.cosh,
+    np.sqrt: lambda x: 0.5/np.sqrt(x),
+    np.square: lambda x: 2.0*x,
+    np.tan: lambda x: 1+np.tan(x)**2,
+    np.tanh: lambda x: 1-np.tanh(x)**2}
+
+UFUNC_DERIVATIVES[np.divide] = UFUNC_DERIVATIVES[np.true_divide]
+UFUNC_DERIVATIVES[np.abs] = UFUNC_DERIVATIVES[np.fabs]
+
+if hasattr(np, 'cbrt'):
+    UFUNC_DERIVATIVES[np.cbrt] = lambda x: 1./(3.*np.cbrt(x)**2)
```

Comment:
Had to think again why I did it, but then remembered: `np.cbrt` is pretty optimized, as is squaring. As a result, what I have is faster than just raising to the `2/3`:
```
In [2]: a = np.arange(100.)

In [3]: %timeit np.cbrt(a)
100000 loops, best of 3: 2.93 µs per loop

In [4]: %timeit np.cbrt(a)**2
100000 loops, best of 3: 3.78 µs per loop

In [5]: %timeit a**(2/3)
100000 loops, best of 3: 7.33 µs per loop
```